src/tests/end-to-end/common/page-controllers/details-view-page.ts

```diff
@@ -122,14 +123,26 @@ export class DetailsViewPage extends Page {
         await this.clickSelector(overviewSelectors.exportDropdown);
     }
 
-    public async downloadExportReport(selector: string): Promise<string> {
+    public async downloadExportReport(
+        selector: string,
+        saveAsFileName?: string,
+    ): Promise<Download> {
         const [download] = await Promise.all([
             this.underlyingPage.waitForEvent('download'),
             this.clickSelector(selector),
         ]);
-        const suggestedName = download.suggestedFilename();
+        if (saveAsFileName !== undefined) {
+            await download.saveAs(saveAsFileName);
+        }
+        return download;
+    }
+
+    public async getDownloadFileName(download: Download) {
```

Comment:
I think the only test that uses this would probably read a bit more clearly if we just inlined this helper - when I first read `getDownloadFileName` in the context of `overview.test.ts`, I wondered if this meant "the filename our extension suggested" vs "the generated filename Playwright actually used". I think it's more obvious that the test is testing the right thing to have it read `suggestedFilename`.